libraries/AC_PrecLand/AC_PrecLand_MAVLink.cpp

```diff
@@ -0,0 +1,55 @@
+#include "AC_PrecLand_config.h"
+
+#if AC_PRECLAND_MAVLINK_ENABLED
+
+#include "AC_PrecLand_MAVLink.h"
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+
+// perform any required initialisation of backend
+void AC_PrecLand_MAVLink::init()
+{
+    // set healthy
+    _state.healthy = true;
+}
+
+// retrieve updates from sensor
+void AC_PrecLand_MAVLink::update()
+{
+    _los_meas.valid = _los_meas.valid && AP_HAL::millis() - _los_meas.time_ms <= 1000;
+}
+
+void AC_PrecLand_MAVLink::handle_msg(const mavlink_landing_target_t &packet, uint32_t timestamp_ms)
+{
+    _distance_to_target = packet.distance;
```

Comment:
We should probably have a `_los_meas.valid = false` here, to prevent stale usage of `_los_meas.vec_unit` incase this function returns early? We check the timestamp of the last consumed packet to make sure we don't consume duplicates, so this is probably not a problem yet, but my worry is that someone may use this in the future by mistake elsewhere.

Reply:
I think the other backends may need this as well

Reply:
Hi @rishabsingh3003,

Thanks very much for the review, very few others are familiar with this library.

Re adding specifically setting "_los_meas.valid = false", I've just done a light review and I don't think this is necessary because the target's position will only be updated when "_los_meas.time_ms" is changed by a backend and each backend's update() function includes a timeout.  Put another way, I don't think we need to invalidate the last known reading before the timeout does.